WordPress/wordpress-playground
# Experimental: Adds declarative properties to the Blueprint schema

## What is this PR doing?

First stab at https://github.com/WordPress/blueprints/issues/33

This PR proposes a first few declarative Blueprint properties:

```js
{
	"landingPage": "/wp-admin/post-new.php",
	"plugins": [ 
		"gutenberg", 
		"classic-editor",
		"https://mysite.com/plugin.zip",
		{ "resource": "url", "https://mysite.com/plugin.zip" }
	],
	"login": true,
	"siteOptions": {
		"blogname": "My site!"
	}
}
```

Under the hood, all these declarative properties are translated into `steps` and prepended to the list of Blueprint steps. Therefore, the imperative `steps` API is still supported and can be mixed with the declarative syntax.

These new options are all marked as deprecated from the start to discourage their use. Once they're stable, I'll remove the deprecation notice.

## Testing instructions

* Pull this branch
* Paste the below Blueprint at http://localhost:5400/website-server/# <-- here
* Confirm it worked as expected

```js

```js
{
	"landingPage": "/wp-admin/post-new.php",
	"plugins": [ 
		"gutenberg", 
		"classic-editor"
	],
	"login": true,
	"siteOptions": {
		"blogname": "My site!"
	}
}
```



cc @dmsnell @seanmorris
